2.0/s2i/bin/assemble

```
@@ -31,20 +31,24 @@ if [ -n "${DOTNET_NPM_TOOLS}" ]; then
   popd
 fi
 
-# project.json defaults to naming the assembly the same as the folder
-if [ -n "$DOTNET_ASSEMBLY_NAME" ]; then
-  mkdir "$DOTNET_ASSEMBLY_NAME"
-  cd "$DOTNET_ASSEMBLY_NAME"
-fi
-
 echo "---> Copying application source ..."
 cp -Rf /tmp/src/. ./
 
 # output assembly name
 if [ -n "$DOTNET_ASSEMBLY_NAME" ]; then
   APP_DLL_NAME="${DOTNET_ASSEMBLY_NAME}.dll"
 else
-  APP_DLL_NAME="$(basename "$(realpath "${DOTNET_STARTUP_PROJECT}")").dll"
+  CSPROJFILES=(`find "${DOTNET_STARTUP_PROJECT}" -maxdepth 1 -name "*.csproj"`)
+  if [ ${#CSPROJFILES[@]} -eq 1 ]; then
+    CSPROJFILE=$(basename ${CSPROJFILES[0]})
+    APP_DLL_NAME="${CSPROJFILE: : -7}.dll"
+  elif [ ${#CSPROJFILES[@]} -eq 0 ]; then
```

Maybe we should do these checks outside of this scope? I say that because having 1 (and only 1) csproj file is required even if the assembly name is defined, but as it stands here these messages won't be seen and the script won't error out until much later.

So maybe move the check for 1 (and only 1) csproj file up before the npm stuff (line 11) and fail early in this case, then we don't have to worry about checking it here.